File: HPC/Organized/archive/hw02/py3/numint.py
#!/usr/bin/python3

import math # need for pi, cos
import sys
import time

# define a, b
a = 0
b = math.pi / 2

# get resolution: number of subintervals
N = int(sys.argv[1])
#N = int(input('N: '))

step = (b - a) / N
step_half = step / 2
res = 0;

start = time.time()

for i in range(1, N):
    if i % 2==0:
        res += 2*math.cos(i*step)
    else:
        res += 4*math.cos(i*step)

res = (step/3)*(math.cos(a)+math.cos(b)+res)

end = time.time()
    
analytic = math.sin(b)
print(str(abs(analytic-res)) + '\t' + str(end-start))

File: HPC/Organized/archive/hw02/c3/numint.c
#include <stdio.h>
#include <math.h>
#include <sys/time.h>

double gettime() {
    struct timeval tv;
    gettimeofday( & tv, NULL);
    return tv.tv_sec + 0.000001 * tv.tv_usec;
}

int main(int argc, char ** argv) {
    double a = 0;
    double b = M_PI / 2;
    double N = (double)atoll(argv[1]);
    //double N = 0.0;
    //scanf("%lf", & N);

    //trapezoid formula
    double step = (b - a) / N;
    double step_half = step / 2;
    double res = 0;

    double start = gettime();
    double i = 0;
    for (i = 1; i < N; i++) {
        if ((int) i % 2 == 0) {
            res += 2 * cos(i * step);
        } else {
            res += 4 * cos(i * step);
        }

    }
    res = (step / 3) * (cos(a) + cos(b) + res);
    double end = gettime();

    double analytic = sin(b);
    printf("%16.18f\t", fabs(analytic - res));
    printf("%f\n", end - start);

    return 0;
}